packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.test.tsx

```diff
@@ -0,0 +1,325 @@
+  { key: '1', text: '1' },
+  { key: '2', text: '2' },
+  { key: '3', text: '3' }
+];
+
+const DEFAULT_OPTIONS2: ISelectableOption[] = [
+  { key: '1', text: 'One' },
+  { key: '2', text: 'Foo' },
+  { key: '3', text: 'Bar' }
+];
+
+describe('ComboBox', () => {
+
+  it('Can flip between enabled and disabled.', () => {
+    let wrapper = shallow(
+      <ComboBox
+        label='testgroup'
+        options={ DEFAULT_OPTIONS }
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+
+    expect(comboBoxRoot.find('.is-disabled').length).equals(0, `shouldn't be disabled`);
+    expect(comboBoxRoot.find('[data-is-interactable=true]').length).equals(1, 'data-is-focusable="true"');
+
+    wrapper = shallow(
+      <ComboBox
+        disabled={ true }
+        label='testgroup'
+        options={ DEFAULT_OPTIONS }
+      />);
+    comboBoxRoot = wrapper.find('.ms-ComboBox');
+
+    expect(comboBoxRoot.find('.is-disabled').length).equals(1, `should be disabled`);
+    expect(comboBoxRoot.find('[data-is-interactable=false]').length).equals(1, 'data-is-focusable="false"');
+  });
+
+  it('Renders no selected item in default case', () => {
+
+    let wrapper = mount(
+      <ComboBox
+        label='testgroup'
+        options={ DEFAULT_OPTIONS }
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+    let inputElement = comboBoxRoot.find('[role="combobox"]');
+
+    expect(inputElement.text()).equals('');
+  });
+
+  it('Renders a selected item in uncontrolled case', () => {
+    let wrapper = mount(
+      <ComboBox
+        label='testgroup'
+        defaultSelectedKey='1'
+        options={ DEFAULT_OPTIONS }
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+    let inputElement = comboBoxRoot.find('input');
+
+    expect(inputElement.props().value).equals('1');
+  });
+
+  it('Renders a selected item in controlled case', () => {
+    let wrapper = mount(
+      <ComboBox
+        label='testgroup'
+        selectedKey='1'
+        options={ DEFAULT_OPTIONS }
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+    let inputElement = comboBoxRoot.find('input');
+
+    expect(inputElement.props().value).equals('1');
+  });
+
+  it('Renders a default value with options', () => {
+    let wrapper = mount(
+      <ComboBox
+        label='testgroup'
+        value='1'
+        options={ DEFAULT_OPTIONS }
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+    let inputElement = comboBoxRoot.find('input');
+
+    expect(inputElement.props().value).equals('1');
+  });
+
+  it('Renders a default value with no options', () => {
+    let wrapper = mount(
+      <ComboBox
+        label='testgroup'
+        value='1'
+      />);
+    let comboBoxRoot = wrapper.find('.ms-ComboBox');
+    let inputElement = comboBoxRoot.find('input');
+
+    expect(inputElement.props().value).equals('1');
+  });
+
+  it('Can change items in uncontrolled case', () => {
+    let comboBoxRoot;
+
+    try {

```

Comment:
Can we remove the try/finally's? I don't think they are needed...if the test throws...it should fail and the finally is not doing any clean up or similar.